apps/pp/views/reference_feedbacks.py

```diff
@@ -0,0 +1,110 @@
+                                                 **{self.resource_attr: True})
+        except IntegrityError:
+            return ErrorResponse('Failed to create object')
+        return Response(data=None)
+
+    @swagger_auto_schema(operation_description=deprecated_description)
+    @method_decorator(allow_lazy_user)
+    def delete(self, request, reference_id):
+        try:
+            model = UserReferenceFeedback.objects.get(reference_id=reference_id, user=request.user,
+                                                      **{self.resource_attr: True})
+        except UserReferenceFeedback.DoesNotExist:
+            return NotFoundResponse()
+
+        model.delete()
+        return Response(data=None)
+
+    @swagger_auto_schema(responses={200: FeedbackSerializer})
+    @method_decorator(allow_lazy_user)
+    def get(self, request, reference_id):
+        try:
+            feedback = UserReferenceFeedback.objects.get(reference_id=reference_id, user=request.user,
+                                                         **{self.resource_attr: True})
+        except UserReferenceFeedback.DoesNotExist:
+            return NotFoundResponse('Resource not found')
+
+        pre_serializer = DataPreSerializer(feedback, {'attributes': feedback})
+        pre_serializer.set_relation(resource_name=get_resource_name(feedback, related_field='reference_id'),
+                                    resource_id=feedback.reference_id)
+        return Response(self.serializer_class(pre_serializer.data, context={'request': request}).data)
+
+
+class FeedbackSingle(APIView):
+    resource_attr = None
+    serializer_class = FeedbackSerializer
+
+    @swagger_auto_schema(responses={200: FeedbackSerializer})
+    @method_decorator(allow_lazy_user)
+    def get(self, request, feedback_id):
+        try:
+            feedback = UserReferenceFeedback.objects.get(id=feedback_id, user=request.user,
+                                                         **{self.resource_attr: True})
+        except UserReferenceFeedback.DoesNotExist:
+            return NotFoundResponse('Resource not found')
+
+        pre_serializer = DataPreSerializer(feedback, {'attributes': feedback})
+        pre_serializer.set_relation(resource_name=get_resource_name(feedback, related_field='reference_id'),
+                                    resource_id=feedback.reference_id)
+        return Response(self.serializer_class(pre_serializer.data, context={'request': request}).data)
+
+    @method_decorator(allow_lazy_user)
+    def delete(self, request, feedback_id):
+        try:
+            feedback = UserReferenceFeedback.objects.get(id=feedback_id, user=request.user,
+                                                         **{self.resource_attr: True})
+        except UserReferenceFeedback.DoesNotExist:
+            return NotFoundResponse()
+
+        feedback.delete()
+        return Response()
+
+
+class FeedbackList(APIView):
```

Comment:
`List`? This is a mistake, right?

Reply:
Ok, I think I understand. Everything is POSTed to the general "collection" URL? If that's the point, I'd consider using a bit more intuitive name. Maybe something like "ResourceView" (not too good either, but I'm just pointing the direction)? Maybe JSON API docs have given the `/<resource>/` view a name?